### PR TITLE
Add current_component_host helper

### LINKED ISSUE #197: Courses/controller_spec references the Module Host

### PR DESCRIPTION
**Pull request by allenwq**
Also fixes #197

### CODE REVIEW

File: app/controllers/course/controller.rb [hunk 1]
```diff
@@ -54,4 +54,14 @@ def current_course_user
...
+
+  # Gets the component host for current instance and course
+  #
+  # @return [Course::ComponentHost] The instance of component host using settings from instance and
+  #                                 course
+  def current_component_host
+    @current_component_host ||= Course::ComponentHost.new(current_tenant.settings,
+                                                          current_course.settings)
+  end
+  helper_method :current_component_host
```
Comment (lowjoel) [inline app/controllers/course/controller.rb]: not a helper method. this should not be accessible to views.

File: spec/controllers/course/controller_spec.rb [hunk 2]
```diff
@@ -64,5 +50,28 @@ def publicly_accessible?
         end
       end
     end
+
+    describe '#current_component_host' do
+      it 'returns the current course' do
```
Comment (lowjoel) [inline spec/controllers/course/controller_spec.rb]: description doesn't match.

Comment (allenwq) [inline spec/controllers/course/controller_spec.rb]: Due to copy paste >_<

File: spec/controllers/course/controller_spec.rb [hunk 3]
```diff
@@ -64,5 +50,28 @@ def publicly_accessible?
...
+        allow(controller).to receive(:current_course).and_return(course)
```
Comment (lowjoel) [inline spec/controllers/course/controller_spec.rb]: I'm not sure why you need to stub so many things.

Comment (allenwq) [inline spec/controllers/course/controller_spec.rb]: I'm trying to isolate it from other tests here, make it not depend on <code>current_course</code>

**Pull request merged**